wardrobe/tradable_kitties/src/tests.rs

```diff
@@ -0,0 +1,864 @@
+    //  let input_coin2 = Coin::<0>(90);
+    let output_coin = Coin::<0>(100);
+
+    let result = TradableKittyConstraintChecker::<0>::Buy.check(
+        &[input_kitty.into(), input_coin1.into()],
+        &[],
+        &[output_kitty.into(), output_coin.into()],
+    );
+    assert_eq!(
+        result,
+        Err(TradeableKittyError::InsufficientCollateralToBuyKitty)
+    );
+}
+
+#[test]
+fn buy_kitty_coin_output_value_exceeds_input_coin_value_fails() {
+    let mut input_kitty = TradableKittyData::default_tradable_kitty();
+    input_kitty.price = Some(101);
+    let output_kitty = input_kitty.clone();
+
+    let input_coin1 = Coin::<0>(100);
+    let input_coin2 = Coin::<0>(90);
+    let output_coin = Coin::<0>(300);
+
+    let result = TradableKittyConstraintChecker::<0>::Buy.check(
+        &[input_kitty.into(), input_coin1.into(), input_coin2.into()],
+        &[],
+        &[output_kitty.into(), output_coin.into()],
+    );
+    assert_eq!(
+        result,
+        Err(TradeableKittyError::MoneyError(
+            MoneyError::OutputsExceedInputs
+        ))
+    )
+}
+
+#[test]
+fn buy_kitty_input_zero_coin_value_fails() {
+    let mut input_kitty = TradableKittyData::default_tradable_kitty();
+    input_kitty.price = Some(101);
+    let output_kitty = input_kitty.clone();
+
+    let input_coin1 = Coin::<0>(0);
+    let input_coin2 = Coin::<0>(90);
+    let output_coin = Coin::<0>(300);
+
+    let result = TradableKittyConstraintChecker::<0>::Buy.check(
+        &[input_kitty.into(), input_coin1.into(), input_coin2.into()],
+        &[],
+        &[output_kitty.into(), output_coin.into()],
+    );
+    assert_eq!(
+        result,
+        Err(TradeableKittyError::MoneyError(MoneyError::ZeroValueCoin))
+    )
+}
+
+#[test]
+fn buy_kitty_output_zero_coin_value_fails() {
+    let mut input_kitty = TradableKittyData::default_tradable_kitty();
+    input_kitty.price = Some(101);
+    let output_kitty = input_kitty.clone();
+
+    let input_coin1 = Coin::<0>(100);
+    let input_coin2 = Coin::<0>(90);
+    let output_coin = Coin::<0>(0);
+
+    let result = TradableKittyConstraintChecker::<0>::Buy.check(
+        &[input_kitty.into(), input_coin1.into(), input_coin2.into()],
+        &[],
+        &[output_kitty.into(), output_coin.into()],
+    );
+    assert_eq!(
+        result,
+        Err(TradeableKittyError::MoneyError(MoneyError::ZeroValueCoin))
+    )
+}
+
+#[test]
+fn buy_kitty_basic_kitty_fails() {
+    let input_kitty = TradableKittyData::default_kitty();
+    let output_kitty = input_kitty.clone();
+
+    let input_coin1 = Coin::<0>(100);
+    let input_coin2 = Coin::<0>(90);
+    let output_coin = Coin::<0>(0);
+
+    let result = TradableKittyConstraintChecker::<0>::Buy.check(
+        &[input_kitty.into(), input_coin1.into(), input_coin2.into()],
+        &[],
+        &[output_kitty.into(), output_coin.into()],
+    );
+    assert_eq!(result, Err(TradeableKittyError::BadlyTyped))
+}
```

Comment:
In some sense this is a duplicate of your `buy_kitty_wrong_input_type_fails` test. But I actually kind of like having a dedicated case for this non tradable kitty data one. Never hurts to be thorough.

Reply:
I kept it as it is.